apps/web/src/store/attachment.ts

```diff
@@ -0,0 +1,89 @@
+import { Localstorage } from 'data/storage';
+import { del, get, set } from 'idb-keyval';
+import type { Attachment } from 'xmtp-content-type-remote-attachment';
+import { create } from 'zustand';
+import { persist } from 'zustand/middleware';
+
+interface AttachmentState {
+  loadedAttachmentURLs: string[];
+  addLoadedAttachmentURL: (attachmentURL: string) => void;
+}
+interface AttachmentCache {
+  cachedAttachments: Map<string, Attachment>;
+  cacheAttachment: (url: string, attachment: Attachment) => void;
+}
+
+function replacer(key: string, value: any) {
+  if (value instanceof Map) {
+    return {
+      dataType: 'Map',
+      value: [...value]
+    };
+  } else {
+    return value;
+  }
+}
+
+function reviver(key: string, value: any) {
+  if (typeof value === 'object' && value !== null && value.dataType === 'Map') {
+    return new Map(value.value);
+  }
+
+  return value;
+}
+
+export const useAttachmentCacheStore = create(
+  persist<AttachmentCache>(
+    (set) => ({
+      cachedAttachments: new Map<string, Attachment>(),
+      cacheAttachment: (url: string, attachment: Attachment) => {
+        set((state) => {
+          const { cachedAttachments } = state;
+          cachedAttachments.set(url, attachment);
+          return { cachedAttachments };
+        });
+      }
+    }),
+    {
+      name: Localstorage.AttachmentCache,
```

Comment:
We need to create a new storage called IndexDB instead of Localstorage enum

Reply:
I addressed all the other feedback except for this one because it seems like a bigger deal since we use Localstorage for messages right now too. Is there any chance you can either help us with this or is it ok to keep as-is? Fwiw I added clearing of the attachment cache to the other places that we clear the message store! I noticed we were missing that in this PR.